ts/credentialproviderchain.ts

```diff
@@ -0,0 +1,10 @@
+import {CredentialProviderChain} from '../lib/credentials/credential_provider_chain';
+
+const providerChain = new CredentialProviderChain([]);
+
+providerChain.resolvePromise().then(
+    () => {},
```

Comment:
You should make a type assertion on the resolved value here so that the compiler verifies the parameterized promise.